From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 20 Sep 2026 14:50:05 -0500
Subject: [PATCH] meson: Fix missing symbols causing issues with Sushi

By default gnome.compile_resources annotates generated functions with `G_GNUC_INTERNAL` which effectively expands out to `__attribute__((visibility("hidden")))` and causes the symbols to be marked as local functions. Thus when Linux distributions strip the built shared libraries with `--strip-unneeded` these symbols are removed as nothing in libppsview uses them.

Use the `export` argument to have meson instead mark these symbols as global so they are preserved after stripping.

Fixes https://gitlab.gnome.org/GNOME/sushi/-/work_items/172
---
 libview/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libview/meson.build b/libview/meson.build
index 1a3182aac..4103e9d19 100644
--- a/libview/meson.build
+++ b/libview/meson.build
@@ -155,6 +155,7 @@ sources += gnome.compile_resources(
   'libview.gresource.xml',
   source_dir: data_dir,
   c_name: 'pps_view',
+  export: true,
   dependencies: blueprints,
 )
 
